### DISCUSSION TITLE
PropertyGrid 组件绑定复杂对象时

### DISCUSSION THREAD

**Discussion by qianwang09** [HandyOrg/HandyControl#1301] [Q&A] [answered]
PropertyGrid 组件绑定对象是复杂对象，子属性也是对象时 好像默认绑定文本框？如何能嵌套绑定子属性对象

**Comment by NaBian** [comment, marked as answer]
需要你自定义编辑器

**Reply by qianwang09** [reply]
简单定义了一个编辑器把propertyGrid又循环嵌套了一遍，只是多出了子属性对象的搜索框
public class ObjectPropertyEditor : PropertyEditorBase
    {
        public override FrameworkElement CreateElement(PropertyItem propertyItem) => new PropertyGrid
        {
            IsEnabled = !propertyItem.IsReadOnly,
            
        };
        public override DependencyProperty GetDependencyProperty() => PropertyGrid.SelectedObjectProperty;
    }

**Comment by wwangsm** [comment]
这样的样式比较丑，能否把这一块优化，支持多层嵌套
